meson_options.txt

```diff
@@ -71,3 +71,9 @@ option(
   description : 'install zsh completion script in this directory',
   value : '',
 )
+option(
+  'warn_implicit_session_opt',
+  type : 'boolean',
+  description : 'warn when the new terminal session creation behaviour isn\'t specified explicitly',
```

Comment:
```suggestion
  description : 'warn if the new terminal session creation behaviour isn\'t specified explicitly',
```

or even

```suggestion
  description : 'warn if neither --new-session nor --no-new-session is used',
```

Reply:
I personally think it would be premature to emit a warning when the option was only added just now: if we want to eventually swap the default (or add debhelper-style compat levels as in #455, or whatever) then we should give dependent projects like Flatpak at least a little time to start adding `--no-new-session` before we start issuing warnings, but they can't start on that until `--no-new-session` is in a release and reasonably widespread.

I also think whether we emit a warning should be upstream policy rather than a distributor build option: after we think bubblewrap versions with this option are sufficiently widespread, we should emit the warning for everyone, but until then, we should not emit the warning for anyone. Expecting downstreams to make this decision seems like a way to make the intended behaviour of any given version of bubblewrap less clear (and adding the option involves quite a lot of build system boilerplate, particularly until we delete the Autotools build system).

I'd be interested to hear what the other bubblewrap maintainers think about this, though.

Reply:
> I personally think it would be premature to emit a warning when the option was only added just now

I very much agree with this, that is why the option in question is off by default in both `meson_options.txt` and `configure.ac`.

> I also think whether we emit a warning should be upstream policy rather than a distributor build option

To me that sounds like a great idea, in that case i'd assume that it would be better to remove all the `#ifdef`s and build options and just have the warning `#if 0`ed for now. However, my question then is that if you seem OK with starting to issue a warning once `--no-new-session` is widespread, wouldn't it be a good idea to mention it `--help` and/or `bwrap.xml` so that everyone knows the intention?

Thank you for reviewing my code by the way, will address all your points once I have time.